### PR TITLE
Remove the FullCLR-only symbol-info related code from EventManager.cs

### PR DESCRIPTION
**Pull request by daxian-dbw**
## PR Summary

Remove the FullCLR-only symbol-info related code from EventManager.cs, because the `AssemblyBuilder` family types do not support emitting symbol information in .NET Core.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [ ] Not Applicable or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [x] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

File: src/System.Management.Automation/engine/EventManager.cs [hunk 1]
```diff
@@ -673,22 +664,10 @@ private void ProcessNewSubscriber(PSEventSubscriber subscriber, Object source, s
 
             if (_eventAssembly == null)
             {
-#if !CORECLR
-                // Define the assembly that will hold our event handlers
-                StackFrame callStack = new StackFrame(0, true);
-                debugMode = (callStack.GetFileName() != null);
```
Comment (daxian-dbw) [inline src/System.Management.Automation/engine/EventManager.cs]: Note: `callstack.GetFileName()` return null if the corresponding `.pdb` file is not put beside the `.dll` file. So this is actually to see if we are currently debugging the `.dll`.

File: src/System.Management.Automation/engine/EventManager.cs [hunk 2]
```diff
@@ -673,22 +664,10 @@ private void ProcessNewSubscriber(PSEventSubscriber subscriber, Object source, s
...
                 _eventAssembly = AssemblyBuilder.DefineDynamicAssembly(
                     new AssemblyName("PSEventHandler"),
                     AssemblyBuilderAccess.Run);
-            }
-            if (_eventModule == null)
-            {
-#if CORECLR
                 _eventModule = _eventAssembly.DefineDynamicModule("PSGenericEventModule");
-#else
-                _eventModule = _eventAssembly.DefineDynamicModule("PSGenericEventModule", debugMode);
```
Comment (daxian-dbw) [inline src/System.Management.Automation/engine/EventManager.cs]: Note: this `DefineDynamicModule` doesn't have this overload in .NET Core.

File: src/System.Management.Automation/engine/EventManager.cs [hunk 3]
```diff
@@ -748,14 +727,7 @@ private void ProcessNewSubscriber(PSEventSubscriber subscriber, Object source, s
                         }
                     }
                 }
-#if !CORECLR
-                // If it is a ManagementEventWatcher, enable it
-                ManagementEventWatcher eventWatcher = source as ManagementEventWatcher;
```
Comment (daxian-dbw) [inline src/System.Management.Automation/engine/EventManager.cs]: Note: WMIv1 (`System.Management`) is not available in .NET Core by default.

File: src/System.Management.Automation/engine/EventManager.cs [hunk 4]
```diff
@@ -1499,35 +1446,19 @@ private Type GenerateEventHandler(MethodInfo invokeSignature)
...
-
-#if !CORECLR
-            if (debugMode)
-            {
-                argsBuilder.SetLocalSymInfo("args");
+            // Declare a local variable of the type 'object[]' at index 0, say 'object[] args'
+            methodContents.DeclareLocal(typeof(object[]));
 
-                //     new Object[ invokeSignature.GetParameters().Length ]
-                methodContents.MarkSequencePoint(doc, callStack.GetFileLineNumber() - 1, 1, callStack.GetFileLineNumber(), 100);
```
Comment (daxian-dbw) [inline src/System.Management.Automation/engine/EventManager.cs]: Note: API doesn't exist in .NET Core.

File: src/System.Management.Automation/engine/EventManager.cs [hunk 5]
```diff
@@ -1499,35 +1446,19 @@ private Type GenerateEventHandler(MethodInfo invokeSignature)
...
-            }
-#endif
             methodContents.Emit(OpCodes.Ldc_I4, parameterCount);
             methodContents.Emit(OpCodes.Newarr, typeof(Object));
 
-            // Retrieve the args variable from local variable index 0
+            // Store the new array to the local variable 'args'
```
Comment (rjmholt) [inline src/System.Management.Automation/engine/EventManager.cs]: Was this an error in the previous comment?

**Review by rjmholt — APPROVED** [review]
Looks good to me

File: src/System.Management.Automation/engine/EventManager.cs (hunk 5 above)
Comment (rjmholt) [reply src/System.Management.Automation/engine/EventManager.cs]: Oh, I see -- never mind

**Review by TravisEz13 — APPROVED** [review]

**Pull request merged**